pageserver/src/waldecoder.rs

```diff
@@ -126,23 +131,37 @@ impl WalStreamDecoder {
                     return Ok(None);
                 }
 
-                // read xl_tot_len FIXME: assumes little-endian
+                // peek xl_tot_len at the beginning of the record.
+                // FIXME: assumes little-endian
                 self.startlsn = self.lsn;
-                let xl_tot_len = self.inputbuf.get_u32_le();
+                let xl_tot_len = u32::from_le_bytes(self.inputbuf[0..4].try_into().unwrap());
```

Comment:
Consider using the Buf impl on slice to do this:

`self.inputbuf.as_ref().get_u32_le()`
or
`self.inputbuf[..].get_u32_le()`

I believe these should get us the u32 without moving the inputbuf pointer